### PR TITLE
doc: Improve Programming software - STM32CubeIDE

### PR DESCRIPTION
**Pull request by pradeeka7**
#### Summary

The current page (https://www.genericnode.com/docs/getting-started/programming-software/st-examples/stcubeide/) has a lack of information about running a project. This pull request includes some additional information that can be helpful to users when they running a project.

#### Screenshots

image

#### Changes

- Explained how the compiled program cab be flashed to GNSE by running a project
- Explained how to view serial data
- Explained how to reset GNSE
- Provided link to drag and drop flash programming
- Added some screen captures

#### Notes for Reviewers

@elsalahy 

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md [hunk 1]
```diff
@@ -7,10 +7,26 @@ distributions: null
 
 If you are using [STM32CubeIDE](https://www.st.com/en/development-tools/stm32cubeide.html) as your build environment, then using the same tool to program your {{% gnse %}} is convenient.
 
-<!--more-->
```
Comment (elsalahy) [inline doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md:10]: Why remove this line?

This line protects the documentation framework from having all the page information on the thumbnail.

See how this looks now 

Screen Shot 2021-09-13 at 18 01 19

in comparison to the old [overview](https://www.genericnode.com/docs/getting-started/programming-software/st-examples/)

File: doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md [hunk 2]
```diff
@@ -7,10 +7,26 @@ distributions: null
...
 
-1. To program and run or debug {{% gnse %}}, right-click on it and select ***Run As*** or ***Debug As***. When programming, make sure you select **STM32 Cortex-M C/C++ Application** in the **Run As** or **Debug As** section.
+When the [build]({{< ref "/getting-started/se-sw/env-setup/stm32cubeide" >}}) process is done you can flash the binary file to the {{% gnse %}} by running the program from the STM32CubeIDE.
 
-{{< figure src="cubeide.png" alt="cubeide run as" >}}
+Right click on the project name in the **Project Explorer** and select **Run As -> STM32 Cortex-M C/C++ Application**.
+
+{{< figure src="run_as.png" alt="project run as" >}}
+
+This will start to **flash** the compiled program (bin file) to the {{% gnse %}}. The following figure shows the **Console** output, for example, when running the [basic]({{< ref "/applications/se-basic" >}}) project.
```
Comment (elsalahy) [inline doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md:20]: This actually flashes the .elf file as it uses GDB server to do so.
I think there  is no need to mention the file type.

image

**Review by elsalahy — CHANGES_REQUESTED** [review]
Thanks for the contribution @pradeeka7 
Could you please handle the minor recommended fixes?

File: doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md (hunk 1 above)
Comment (pradeeka7) [reply doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md:10]: fixed

File: doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md (hunk 2 above)
Comment (pradeeka7) [reply doc/content/getting-started/programming-software/st-examples/stcubeide/_index.md]: Thanks for the clarification. I thought it was the bin file.
fixed

**Review by elsalahy — APPROVED** [review]
Looks good, will be deployed by the end of the week.

**Pull request merged**